lib/mix/phoenix/schema.ex

```diff
@@ -164,12 +167,14 @@ defmodule Mix.Phoenix.Schema do
   end
 
   @doc """
-  Generates some sample params based on the parsed attributes.
+  Generates some sample params based on the parsed attributes. LiveView cannot be given an invalid
+  value in a select field so we drop enum params for the invalid_live params.
   """
-  def params(attrs, action \\ :create) when action in [:create, :update] do
+  def params(attrs, action \\ :create) when action in [:create, :update, :invalid, :invalid_live] do
```

Comment:
Please take a look at 9a019f607. It solved these problems differently. In any case, `invalid_form_value` already sets it to nil, which is enough to make it invalid, no? Otherwise, I suggest we change the generator to include a select with blank, something like `<%= select f, :foo, prompt: "Choose" %>`.

Reply:
@josevalim The reason I went down the path of separating out enums in live tests is that `render_change` throws an argument error if you try to set anything but one of the valid options in the select field (including trying to set nil):

```
       3) test Index saves new post (DefaultMssqlAppWeb.PostLiveTest)
          test/default_mssql_app_web/live/post_live_test.exs:26
          ** (ArgumentError) value for select "post[status]" must be one of ["unpublished", "published", "deleted"], got: nil
          code: |> render_change() =~ "can&apos;t be blank"
          stacktrace:
            (phoenix_live_view 0.14.7) lib/phoenix_live_view/test/live_view_test.ex:876: Phoenix.LiveViewTest.call/2
            test/default_mssql_app_web/live/post_live_test.exs:36: (test)
```

I discussed this with @chrismccord when this first came up, and the best option we came up with was to simply drop enums from the list of invalid values when dealing with the LiveView tests. It functions correctly on dead views in `phx.gen.html`, so there was no reason to drop them there, hence splitting out `:invalid` and `:invalid_live`. I am open to any suggestions on the path forward.

Reply:
What do you think about my suggestion of adding a prompt to the `select`? This way we can continue submitting blank strings (which is the default invalid value today) and we don't have to change the tests.

Reply:
@josevalim apologies, I had tried using a prompt prior, and then re-tried it after you suggested it, but forgot to mention that the LiveView `render_change` also raises with a prompt present when given a nil value:

```
       1) test Index saves new post (RainyDayWeb.PostLiveTest)
          apps/rainy_day_web/test/rainy_day_web/live/post_live_test.exs:26
          ** (ArgumentError) value for select "post[status]" must be one of ["unpublished", "published", "deleted"], got: ""
          code: |> render_change() =~ "can&apos;t be blank"
          stacktrace:
            (phoenix_live_view 0.14.7) lib/phoenix_live_view/test/live_view_test.ex:876: Phoenix.LiveViewTest.call/2
            test/rainy_day_web/live/post_live_test.exs:36: (test)
```

Reply:
Hrm, that sounds like a bug in LiveView, let me try to reproduce it.